compiler/resolution/functionResolution.cpp

```diff
@@ -3285,6 +3289,11 @@ FnSymbol* resolveNormalCall(CallExpr* call, bool checkonly) {
         // If the first actual is an instance of dtMethodToken and the call is
         // to "init" of a generic record that defined initializers
         resolveInitializer(call);
+        NamedExpr* named = toNamedExpr(call->get(2));
```

Comment:
I meant to move the action which updates the type of the "this" actual into resolveInitializer so it would be consistent for classes and records.  Would that solve your concern, or is the use of the NamedExpr at all still troubling?